packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/AnnotationEditorUtils.ts

```diff
@@ -133,3 +134,17 @@ export function convertToArray<T>(value: T): Array<unknown> {
     return [value]
   }
 }
+
+/**
+ * Returns true if the default behavior of the form should be to live validate
+ * @param existingAnnotations
+ * @param validationSchema
+ */
+export function shouldLiveValidate(
+  existingAnnotations: Record<string, unknown> | undefined,
+  validationSchema: JSONSchema7 | undefined,
+): boolean {
+  const hasExistingAnnotations =
+    existingAnnotations && Object.keys(existingAnnotations).length > 0
+  return Boolean(hasExistingAnnotations && validationSchema)
```

Comment:
If the entity doesn't start with annotations, or if it doesn't have a custom schema, we should not turn on liveValidate (i.e. validate only at the end).